src/test/java/space/obminyashka/items_exchange/controller/UserControllerIntegrationTest.java

```diff
@@ -201,6 +215,24 @@ void setUserAvatar_whenReceivedBMPImage_shouldThrowUnsupportedMediaTypeException
         sendUriAndGetMvcResult(multipart(new URI(USER_SERVICE_CHANGE_AVATAR)).file(bmp), status().isUnsupportedMediaType());
     }
 
+    @Test
+    @WithMockUser(username = "user")
+    void setUserAvatar_whenReceivedSeveralImages_shouldSaveFirstImage() throws Exception {
+        when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
+
+        MockMultipartFile bmp = new MockMultipartFile("image", "image-bmp.bmp", "image/bmp", "image bmp".getBytes());
+        MockMultipartFile jpeg = new MockMultipartFile("image", "test-image.jpeg", MediaType.IMAGE_JPEG_VALUE, "image jpg".getBytes());
+
+        sendUriAndGetMvcResult(multipart(new URI(USER_SERVICE_CHANGE_AVATAR)).file(jpeg).file(bmp), status().is2xxSuccessful());
+
+        Mockito.verify(imageService, times(1)).compress(captor.capture());
+        Mockito.verify(userService, times(1)).findByUsernameOrEmail(any());
+
+        MultipartFile actualImage = captor.getValue();
+
+        assertEquals(jpeg.getName(), actualImage.getName());
```

Comment:
The **compress** method is mocked so you can compare files completely other doing that for their names.